docs/developer-guide/syntax-reference.rst

```diff
@@ -234,18 +234,88 @@ RUN SCRIPT can also be used from the command line, for instance when writing she
+KSQL data types
+===============
+
+KSQL supports the following data types:
+
+Primitive Types
+---------------
+
+KSQL supports the following primitive data types:
+
+-  ``BOOLEAN``
+-  ``INTEGER``
+-  ``BIGINT``
+-  ``DOUBLE``
+-  ``VARCHAR`` (or ``STRING``)
+
+
+Array
+-----
+
+``ARRAY<ElementType>``
+
+.. note:: The ``DELIMITED`` format doesn't support arrays.
+
+KSQL supports fields that are arrays of another type. All the elements in the array must be of the
+same type. The element type can be any valid KSQL type.
+
+You can define arrays within a ``CREATE TABLE`` or ``CREATE STREAM`` statement by using the syntax
+``ARRAY<ElementType>``. For example, ``ARRAY<INT>`` defines an array of integers.
+
+The elements of an array are zero-indexed and can be accessed by using the ``[]`` operator passing
+in the index. For example, ``SOME_ARRAY[0]`` retrieves the first element from the array.
+For more information, see :ref:`operators`.
+
+Map
+---
+
+``MAP<KeyType, ValueType>``
+
+.. note:: The ``DELIMITED`` format doesn't support maps.
+
+KSQL supports fields that are maps. A map has a key and value type. All of the keys must be of the
+same type, and all of the values must be also be of the same type. Currently only ``STRING`` keys are supported. The value type can be any valid KSQL type.
+
+You can define maps within a ``CREATE TABLE`` or ``CREATE STREAM`` statement by using the syntax
+``MAP<KeyType, ValueType>``. For example, ``MAP<STRING, INT>`` defines a map with string keys and
+integer values.
+
+Access the values of a map by using the ``[]`` operator and passing in the key. For example,
+``SOME_MAP['cost']`` retrieves the value for the entry with key ``cost``, or ``null``
+For more information, see :ref:`operators`.
+
+Struct
+------
+
+``STRUCT<FieldName FieldType, ...>``
+
+.. note:: The ``DELIMITED`` format doesn't support structs.
+
+KSQL supports fields that are structs. A struct represents strongly typed structured data. A struct
+is an ordered collection of named fields that have a specific type. The field types can be any valid
+KSQL type.
+
+You can define a structs within a ``CREATE TABLE`` or ``CREATE STREAM`` statement by using the syntax
+``STRUCT<FieldName FieldType, ...>``. For example, ``STRUCT<ID BIGINT, NAME STRING, AGE INT>``
+defines a struct with three fields, with the supplied name and type.
+
+Access the fields of a struct by using the ``->`` operator. For example, ``SOME_STRUCT->ID``
+retrieves the value of the struct's ``ID`` field. For more information, see :ref:`operators`.
+
+
 ===============
 KSQL statements
 ===============
 
 .. tip::
 
-    -  KSQL statements must be terminated with a semicolon (``;``).
-    -  Multi-line statements:
-
-       -  In the CLI you must use a backslash (``\``) to indicate
-          continuation of a statement on the next line.
-       -  Do not use ``\`` for multi-line statements in ``.sql`` files.
+    - KSQL statements must be terminated with a semicolon (``;``).
```

Comment:
this isn't the case for CliSpecificStatements (e.g. `exit` and `clear`) but that's probably okay to omit

Reply:
Yeah, that's probably better called out in the CLI specific bits.